client-runtime/src/main/java/com/microsoft/rest/v2/annotations/EndpointHost.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.rest.v2.annotations;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+
+@Target(value={PARAMETER,METHOD, TYPE})    // The context in which annotation is applicable i.e. this annotation (EndpointHost) can be applied to method, method parameter and interface.
+@Retention(RetentionPolicy.RUNTIME)        // Record this annotation in the class file and make it available during runtime.
+public @interface EndpointHost {
+    String value() default "";
+}
+
+/**
+ * The applicability of this annotation is limited to variable with name {host} in the URI template.
+ *
+ *   When this annotation is applied to a method or interface level, value must be defined statically.
+ *   When this annotation is applied to a method parameter level, the parameter value will be it's value
+ *
+ * TODO: Decide whether we need all three of below or subset of it.
+ *
+ *  [1] Annotation applied to interface level.
+ *
+ *     @EndpointHost("manage.windowsazure.com")
```

Comment:
Also this gets even trickier in the case of multiple environments - I'll send a PR to propose some of my thoughts